libp2p/Host.cpp

```diff
@@ -355,20 +312,42 @@ void Host::runAcceptor()
 				}
 			}
 			
-			if (!success && m_socket->is_open())
+			if (!success && s->is_open())
 			{
 				boost::system::error_code ec;
-				m_socket->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
-				m_socket->close();
+				s->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
+				s->close();
 			}
 
 			m_accepting = false;
+			delete s;
```

Comment:
Yep. ASIO is pre-RAII, doesn't shutdown the socket. Will abstract this to it's own class later when UDP sockets can also be considered, such that there's a cohesive API.